test/test_scripts_run_task.py

```diff
@@ -270,3 +272,56 @@ def _Popen(
 
     assert os.path.isdir(untracked_dir.name) is False
     assert os.path.isfile(untracked_file.name) is False
+
+
+@pytest.fixture(scope="session")  # Tests shouldn't change this repo
+def mock_git_repo():
+    "Mock repository with files, commits and branches for using as source"
+    with tempfile.TemporaryDirectory() as repo:
+        repo_path = str(repo)
+        subprocess.check_call(["git", "init", "-b", "main"], cwd=repo_path)
+
+        def _commit_file(message):
+            untracked_file = tempfile.NamedTemporaryFile(dir=repo, delete=False)
```

Comment:
Should we use a non-random name here, and in the test check that the file exists (if we're on that branch)?